agent/connect/ca/provider_vault_auth_approle.go

```diff
@@ -0,0 +1,60 @@
+package ca
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/hashicorp/consul/agent/structs"
+)
+
+// left out 2 config options as we are re-using vault agent's auth config.
+// Why?
+// remove_secret_id_file_after_reading - don't remove what we don't own
+// secret_id_response_wrapping_path - wrapping the secret before writing to disk
+//                                    (which we don't need to do)
+
+func NewAppRoleAuthClient(authMethod *structs.VaultAuthMethod) (*VaultAuthClient, error) {
```

Comment:
That method doesn't really line up with the use case here. Eg. the NewAppRoleAuth already wants secret_id and role_id as parameters where most of this code is about getting them. It could return one at the end but that would mean it uses different code from the rest, which I think hurts more than it helps.